crates/wascc-provider/src/lib.rs

```diff
@@ -437,21 +498,40 @@ struct VolumeBinding {
 fn wascc_run_http(
     host: Arc<Mutex<WasccHost>>,
     data: Vec<u8>,
-    env: EnvVars,
+    _env: EnvVars,
     volumes: Vec<VolumeBinding>,
     log_path: &Path,
     status_recv: Receiver<ContainerStatus>,
+    port_assigned: i32,
 ) -> anyhow::Result<ContainerHandle<ActorHandle, LogHandleFactory>> {
     let mut caps: Vec<Capability> = Vec::new();
 
+    let mut env_map: HashMap<String, String> = HashMap::new();
+    env_map.insert("PORT".to_string(), port_assigned.to_string());
     caps.push(Capability {
         name: HTTP_CAPABILITY,
         binding: None,
-        env,
+        env: env_map, //replaced _env
     });
     wascc_run(host, data, &mut caps, volumes, log_path, status_recv)
 }
 
+fn find_available_port(port_set: &Arc<Mutex<HashSet<i32>>>) -> i32 {
+    let mut range = rand::thread_rng();
+    let mut port: i32 = -1;
+
+    let mut empty_port: HashSet<i32> = HashSet::new();
+    while empty_port.len() < 2768 {
+        port = range.gen_range(30000, 32768);
+        empty_port.insert(port);
+        if !port_set.lock().unwrap().contains(&port) {
```

Comment:
do we want to handle cases such as when another process running outside of Krustlet is bound to this port, we fail here, or do we consider this an acceptable use case and error out as we do in #288?